### DISCUSSION TITLE
cursor blink

### DISCUSSION THREAD

**Discussion by mark-summerfield** [Textualize/textual#2395] [Q&A] [answered]
Is it possible to switch off cursor blink for a textual application?
Some people (including me) can't work with blinking cursors.
See, for example, https://jurta.org/en/prog/noblink

**Comment by davep** [comment]
Do you mean specifically within an `Input` widget, or more generally in the application? If the former, [see the `cursor_blink` reactive attribute](https://textual.textualize.io/widgets/input/#reactive-attributes); if the latter, that will likely be a problem with your environment because Textual turns off the terminal cursor.

**Reply by mark-summerfield** [reply]
I already have cursor blink off in my xterm (xfce4-terminal) but in the demo (`python3 -m textual`) when I click the Username line editor the cursor is blinking which makes it unusable for me.
I lookup up the docs you linked to and modified the input example:
```python
from textual.app import App, ComposeResult
from textual.widgets import Input

class InputApp(App):
    def compose(self) -> ComposeResult:
        yield Input(placeholder="First Name", cursor_blink=False)
        yield Input(placeholder="Last Name", cursor_blink=False)

if __name__ == "__main__":
    app = InputApp()
    app.run()
```
But this produces errors:
```
TypeError: InputApp(title='InputApp', classes={'-dark-mode'}) compose() method returned an invalid result; __init__() got an unexpected keyword argument 'cursor_blink'
```
Oddly, both `placeholder` _and_ `cursor_blink` are listed as "reactive attributes" https://textual.textualize.io/widgets/input/#__tabbed_1_2 but only the former seems to be accepted.
Mind you, what I'd prefer is to be able to switch off cursor blink globally for the whole application.

**Reply by davep** [reply]
Yes, `cursor_blink` is [a reactive attribute](https://textual.textualize.io/widgets/input/#reactive-attributes), but not a parameter of `__init__`; [`placeholder` is both](https://textual.textualize.io/widgets/input/#textual.widgets.Input).

There isn't currently a single global setting for `Input` blink though.

**Comment by mark-summerfield** [comment, marked as answer]
Thanks for that info. So it _is_ possible to stop cursor blink, but has to be done per-widget. Here's a version of the example that does so:
```python
from textual.app import App, ComposeResult
from textual.widgets import Input

class InputApp(App):
    def compose(self) -> ComposeResult:
        nameInput = Input(placeholder="First Name")
        nameInput.cursor_blink = False
        yield nameInput
        nameInput = Input(placeholder="Last Name")
        nameInput.cursor_blink = False
        yield nameInput

if __name__ == "__main__":
    app = InputApp()
    app.run()
```

**Reply by davep** [reply]
Another approach would be to set the default value for `Input.cursor_blink` on application startup, so all `Inputs` would be created with that setting. For example:

```python
from textual.app import App, ComposeResult
from textual.widgets import Input

class InputApp(App):
    def compose(self) -> ComposeResult:
        yield Input(placeholder="First Name")
        yield Input(placeholder="Last Name")

if __name__ == "__main__":
    Input.cursor_blink = False
    app = InputApp()
    app.run()
```

**Reply by willmcgugan** [reply]
I'm not apposed to adding an env var to disable blinking. We're planning something similar for animation.
